src/panels/lovelace/special-rows/hui-divider-row.ts

```diff
@@ -21,10 +24,6 @@ class HuiDividerRow extends LitElement implements EntityRow {
     }
 
     this._config = {
-      style: {
-        height: "1px",
-        "background-color": "var(--secondary-text-color)",
-      },
```

Comment:
Put these default styles in a class, `style` overrules `class`

Reply:
that works, but I still need user styles to be applied to the `:after` and `:before` if there is a label. I don't think that is possible it is just applied to the full element. So if you set the background color, it is applied to the full length.

Reply:
It still works the same as long as you don't add a label, as in not breaking?

For the before/after styles you could add a style block in the render.

Reply:
right, if you try to style the background color it will break the label.

i don't follow on the render suggestion.

Reply:
If you add a style block in the render method:
```js
render() {
return html`
   <style>      
      .text-divider:before,
      .text-divider:after {
        background: red;
      }
   </style>`
}
```
It will work.

Reply:
So you could do:
```js
for (let [key, value] of Object.entries(this._config.style)) {
  `${key}: ${value};`;
}
```

Reply:
Sorry, I got what you meant by the style block in render, but don't understand what that solves? The current method for applying user-specified styles is to do so on the element. So if they `background: red` it is applied to the full element regardless of if a label is there or not. Additionally, some styling should be applied to the before/after while some should be applied to the full element depending on the goal...

Reply:
Be aware btw that this could be breaking, yours to, as javascripts `style.setProperty` uses for instance `backgroundColor` and css/style is `background-color`.

Reply:
So you should add some checks 😄 
The class name should be dynamic ofc, and you would have to check what goes where? Or add a second property, one for styling the text and one for the rest?